src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -68,14 +68,16 @@ public T Value
 
     [DataContract(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/Arrays")]
 #if USE_REFEMIT
-    public struct KeyValue<K, V>
+    public class KeyValue<K, V>
 #else
```

Comment:
This is part of the significant improvement in this PR and benefits mainly the dictionary scenario. Other are the change to use encoding.GetBytes which improve just a few percentage, and caching instance of the GetUninitializedObject to avoid repeated reflection which already improves a lot, I recall close to 50%. With this change the performance is at the same bar or better than Desktop in dictionary.